application/listeners/listeners.go

```diff
@@ -465,9 +466,15 @@ func getID(p events.Payload) (int, error) {
 	return id, nil
 }
 
-func newListenerContext() *listenerContext {
-	ctx := &listenerContext{
+func newListenerContext() listenerContext {
+	ctx := listenerContext{
 		params: map[interface{}]interface{}{},
 	}
+	ctx.DefaultContext = buffalo.DefaultContext{}
+	ctx.DefaultContext.Context = context.Background()
+
+	c2, _ := context.WithCancel(ctx.DefaultContext.Context)
+	ctx.DefaultContext.Context = c2
```

Comment:
`ctx.DefaultContext.Context` is assigned twice. Is the first one unnecessary?

Reply:
I guess not. Since it returned a context, I thought I must need to reassign it.  But it works without it.